### ISSUE TITLE
Test packages are not analyzed

### ISSUE THREAD

**Issue by zimmski** [mgechev/revive#674]
Files of test packages (package name with "_test" suffix) are not analyzed. For me, this is a bug since they are on the same level  as _test.go files which are analyzed.

1. mkdir -p src/foobar.com/bar
2. Add the following file in src/foobar.com/somepackage/baz_test.go

```go
package somepackage

import "errors"

func foo() error {
        return errors.New("hello")
}

func bar() {
        foo()
}
```

3. Do a `revive foobar.com/... and see that an error is reported
4. Change "package somepackage" to "package somepackage_test"
5. Do the revive command again, and see that this time no error is reported but it should.

**Comment by chavacava** [comment]
Hi @zimmski, thanks for filling the issue.
I was not able to reproduce the problem.
I've setup the following directory tree
```
src
└── foobar.com
    ├── somepackage
    │   ├── baz.go
    │   └── baz_test.go
    └── somepackage_test
        ├── baz.go
        └── baz_test.go

```
Every `.go` file has the following content:
```go
package somepackage

import "errors"

func Foo() error {
	return errors.New("hello")
}

func bar() {
	foo()
}
```
(please notice `Foo` with capital `F` to trigger a failure due to comment absence on public function)

I run
```
revive  src/...
```
as indicated in your description, and I get:
```
src/foobar.com/somepackage/baz.go:5:1: exported function Foo should have comment or be unexported
src/foobar.com/somepackage_test/baz.go:5:1: exported function Foo should have comment or be unexported
```
The result is OK. The rule `exported` generates a failure in both `baz.go` files even if one of them is under `somepackage_test`
(please notice that the rule ignores `*_test.go` files)

I'm missing something?
I've used the branch https://github.com/mgechev/revive/tree/reproduce/674 as sandbox
